recipes/boost/all/conanfile.py

```diff
@@ -59,42 +87,67 @@ class BoostConan(ConanFile):
         "extra_b2_flags": "ANY",  # custom b2 flags
         "i18n_backend": ["iconv", "icu", None],
     }
-    options.update({"without_%s" % libname: [True, False] for libname in lib_list})
+    for _name in CONFIGURE_OPTIONS:
+        options["without_{}".format(_name)] = [True, False]
 
     default_options = {
-        'shared': False,
-        'header_only': False,
-        'error_code_header_only': False,
-        'system_no_deprecated': False,
-        'asio_no_deprecated': False,
-        'filesystem_no_deprecated': False,
-        'fPIC': True,
-        'layout': 'system',
-        'magic_autolink': False,
-        'python_executable': 'None',
-        'python_version': 'None',
-        'namespace': 'boost',
-        'namespace_alias': False,
-        'multithreading': True,
-        'zlib': True,
-        'bzip2': True,
-        'lzma': False,
-        'zstd': False,
-        'segmented_stacks': False,
+        "shared": False,
+        "header_only": False,
+        "error_code_header_only": False,
+        "system_no_deprecated": False,
+        "asio_no_deprecated": False,
+        "filesystem_no_deprecated": False,
+        "fPIC": True,
+        "layout": "system",
+        "magic_autolink": False,
+        "python_executable": "None",
+        "python_version": "None",
+        "namespace": "boost",
+        "namespace_alias": False,
+        "multithreading": True,
+        "zlib": True,
+        "bzip2": True,
+        "lzma": False,
+        "zstd": False,
+        "segmented_stacks": False,
         "debug_level": 0,
-        'pch': True,
-        'extra_b2_flags': 'None',
-        "i18n_backend": 'iconv',
+        "pch": True,
+        "extra_b2_flags": "None",
+        "i18n_backend": "iconv",
     }
+    for _name in CONFIGURE_OPTIONS:
+        default_options["without_{}".format(_name)] = False
+    for _name in ("graph_parallel", "mpi", "python",):
+        default_options["without_{}".format(_name)] = True
 
-    for libname in lib_list:
-        if libname != "python":
-            default_options.update({"without_%s" % libname: False})
-    default_options.update({"without_python": True})
     short_paths = True
     no_copy_source = True
     exports_sources = ['patches/*']
 
+    def export(self):
+        self.copy(self._dependency_filename, src="dependencies", dst="dependencies")
+
+    @property
+    def _dependency_filename(self):
+        return "dependencies-{}.yml".format(self.version)
```

Comment:
```suggestion
    @property
    def _dependency_filename(self):
        return "dependencies-{}.yml".format(self.version)

    def export(self):
        self.copy(self._dependency_filename, src="dependencies", dst="dependencies")
```

Reply:
no need to trigger a rebuild just for this, if there's a problem it would be appreciated but I understand the logic so there's no real need.

Reply:
Thanks! I will keep it then.